picard/ui/searchdialog.py

```diff
@@ -0,0 +1,368 @@
+                track=metadata['title'],
+                artist=metadata['artist'],
+                release=metadata['tracknumber'],
+                tnum=metadata['totaltracks'],
+                tracks=metadata['totaltracks'],
+                qdur=str(metadata.length / 2000),
+                isrc=metadata['isrc'],
+                limit=25)
+
+    def handle_reply(self, document, http, error):
+        if error:
+            error_msg = _("Unable to fetch results. Close the dialog and try "
+                    "again. See debug logs for more details.")
+            self.show_error(error_msg)
+            return
+
+        try:
+            tracks = document.metadata[0].recording_list[0].recording
+        except (AttributeError, IndexError):
+            error_msg = _("No results found. Please try a different search query.")
+            self.show_error(error_msg)
+            return
+
+        if self.file_:
+            tmp = sorted((self.file_.orig_metadata.compare_to_track(
+                track, File.comparison_weights) for track in tracks),
+                reverse=True,
+                key=itemgetter(0))
+            tracks = [item[3] for item in tmp]
+
+        del self.search_results[:]  # Clear existing data
+        self.parse_tracks_from_xml(tracks)
+        self.display_results()
+
+    def display_results(self):
+        self.show_table(self.table_headers)
+        for row, obj in enumerate(self.search_results):
+            track = obj[0]
+            table_item = QtGui.QTableWidgetItem
+            self.table.insertRow(row)
+            self.table.setItem(row, 0, table_item(track.title))
+            self.table.setItem(row, 1, table_item(track.length))
+            self.table.setItem(row, 2, table_item(track.artist))
+            self.table.setItem(row, 3, table_item(track.release))
+            self.table.setItem(row, 4, table_item(track.date))
+            self.table.setItem(row, 5, table_item(track.country))
+            self.table.setItem(row, 6, table_item(track.release_type))
+
+    def parse_tracks_from_xml(self, tracks_xml):
+        for node in tracks_xml:
+            rec_id = node.id
+            rec_title = node.title[0].text
+            artist = artist_credit_from_node(node.artist_credit[0])[0]
+            try:
+                length = format_time(node.length[0].text)
+            except AttributeError:
+                length = ""
+            if "release_list" in node.children and "release" in node.release_list[0].children:
+                releases = node.release_list[0].release
+                for release in releases:
+                    rel_id = release.id
+                    rel_title = release.title[0].text
+                    if "date" in release.children:
+                        date = release.date[0].text
+                    else:
+                        date = None
+                    if "country" in release.children:
+                        country = release.country[0].text
```

Comment:
> Wouldn't passing countries through context in ugettext_attr work (as there's no ugettext_countries in i18n.py)? Like ugettext_attr(country, "countries").

According to the code [ugettext_countries is a builtin](https://github.com/metabrainz/picard/blob/a32c0b7be827bd2291b3508dad1a3d064d47cc6b/picard/i18n.py#L95-L95), so you don't need to import anything to use it.

You can't use `ugettext_attr` for countries, since it's not the context that differs. They're actually in completely different domains (i.e. sets of strings). po/attributes/attributes.pot and po/countries/countries.pot are the templates for those domains (you can open those files in a text editor to see all the strings that get translated).

Context is mainly used when a word can produce different translations depending on how it's used. For example, in musicbrainz-server we disambiguate between `l('Series', 'singular')` and `l('Series', 'plural')`, since those words are the same in English but not in other languages.